apps/deploy-web/src/components/billing-usage/TrendIndicator/TrendIndicator.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { isToday } from "date-fns";
+import { GraphDown, GraphUp } from "iconoir-react";
+
+const COMPONENTS = {
+  GraphUp,
+  GraphDown
+};
+
+type HistoryData = Array<{ date: string; [key: string]: number | string | null }>;
+
+type Keys<Data extends HistoryData> = keyof Omit<Data[number], "date">;
+
+export type TrendIndicatorProps<Field extends Keys<Data>, Data extends HistoryData = HistoryData> = {
+  isFetching: boolean;
+  data: Data;
+  field: Field;
+  components?: typeof COMPONENTS;
+};
+
+export const TrendIndicator = <Field extends string & Keys<Data>, Data extends HistoryData = HistoryData>({
+  isFetching,
+  data,
+  field,
+  components: { GraphUp, GraphDown } = COMPONENTS
+}: TrendIndicatorProps<Field, Data>) => {
+  const trendData = React.useMemo(() => {
+    if (data.length < 2) return null;
+
+    const sortedData = data
+      .filter(item => item[field] !== undefined && item[field] !== null)
```

Comment:
doesn't seem like this can be `undefined` looking at the type. I guess if you expect only `number` it's better be checkin that?
```suggestion
      .filter(item => typeof item[field] === 'number')
```

Reply:
Yes, done